### DISCUSSION TITLE
realtime update values from external script

### DISCUSSION THREAD

**Discussion by Nahas-N** [cunnane/xloil#55] [General]
how to change values by running python script externally 

i have read [this ](https://xloil.readthedocs.io/en/stable/xlOil_Python/ExcelApplication.html#excel-automation)

i want to open an existing workbook 

`app = xlo.Application(workbook="test.xlsx")`
but getting this run time error 
RuntimeError: Failed to create Application object from workbook

by loading blank workbook getting an exception of xloil load failed 
`app = xlo.Application()`

i am able to load function from python by embedding way . but not to update with external script 
exact issue is not able to register the xll . 

kindly a walkthrough is helpful . because i believe its some lack of documentation .

**Comment by cunnane** [comment]
`xlo.Application(workbook="test.xlsx")` tries to find the Excel instance which has the `test.xlsx` workbook *already* open.  If you want to start a new instance of Excel then
```
app = xlo.Application()
app.open("test.xlsx")
```
should work.

When you drive Excel externally like this, *xloil.xll* isn't involved, so I'm not certain what your mean about registering the xll?

Copying the [test runner script](https://github.com/cunnane/xloil/blob/master/libs/xlOil_Python/Package/test_SpreadsheetRunner.py) should give you a good start with using xlOil like this.
